pkg/api/dashboard_public.go

```diff
@@ -2,40 +2,43 @@ package api
 
 import (
 	"errors"
+	"fmt"
 	"net/http"
 	"strconv"
+	"strings"
 
 	"github.com/grafana/grafana/pkg/api/dtos"
 	"github.com/grafana/grafana/pkg/api/response"
 	"github.com/grafana/grafana/pkg/models"
 	"github.com/grafana/grafana/pkg/services/dashboards"
+	"github.com/grafana/grafana/pkg/services/datasources"
 	"github.com/grafana/grafana/pkg/web"
 )
 
 // gets public dashboard
 func (hs *HTTPServer) GetPublicDashboard(c *models.ReqContext) response.Response {
-	publicDashboardUid := web.Params(c.Req)[":uid"]
+	accessToken := web.Params(c.Req)[":accessToken"]
 
-	dash, err := hs.dashboardService.GetPublicDashboard(c.Req.Context(), publicDashboardUid)
+	dash, err := hs.dashboardService.GetPublicDashboard(c.Req.Context(), accessToken)
 	if err != nil {
 		return handleDashboardErr(http.StatusInternalServerError, "Failed to get public dashboard", err)
 	}
 
 	meta := dtos.DashboardMeta{
-		Slug:               dash.Slug,
-		Type:               models.DashTypeDB,
-		CanStar:            false,
-		CanSave:            false,
-		CanEdit:            false,
-		CanAdmin:           false,
-		CanDelete:          false,
-		Created:            dash.Created,
-		Updated:            dash.Updated,
-		Version:            dash.Version,
-		IsFolder:           false,
-		FolderId:           dash.FolderId,
-		IsPublic:           dash.IsPublic,
-		PublicDashboardUid: publicDashboardUid,
+		Slug:                       dash.Slug,
+		Type:                       models.DashTypeDB,
+		CanStar:                    false,
+		CanSave:                    false,
+		CanEdit:                    false,
+		CanAdmin:                   false,
+		CanDelete:                  false,
+		Created:                    dash.Created,
+		Updated:                    dash.Updated,
+		Version:                    dash.Version,
+		IsFolder:                   false,
+		FolderId:                   dash.FolderId,
+		IsEnabled:                  true,
```

Comment:
Maybe remove IsEnabled

Reply:
Same as the comment above, it is being used.